C/Bit--Manipulation/Swap_even_and_odd_bits.c

```diff
@@ -0,0 +1,36 @@
+#include <stdio.h>
+
+unsigned int swapBits(unsigned int x)
+{
+    unsigned int evenBits = x & 0xAAAAAAAA;
+    /* 0x is written here. It means it is hexadecimal number and computer stores hexadecimal number in binary format. 
+    Also, A is written here 8 times. So, it means it will store 32 bits integer i.e., (8*4=32) . Each A will be replaced by 1010 . 
+    So, in that way 32 bit integer  will be stored. 
+    */
+
+    /*A in binary is 1010 . So, by performing bitwise & operation, we will get all the even bits of number x since, 1 is 
+   present at even positions. (Masking) */
+
+    unsigned int oddBits = x & 0x55555555; //5 in decimal is 0101. So, by performing bitwise & operation, we will get
+                                           // all the odd bits of number x since 1 is present at odd positions.
+                                           //(Masking)
+
+    evenBits >>= 1; // Right shifting even bits
+    oddBits <<= 1;  // Left shifting odd bits
+
+    return (evenBits | oddBits); // While doing this bitwise OR operation, the adjacent( even and odd bits ) are
+                                 // interchanged
+}
+
+int main()
+{
+    unsigned int x;
+    printf("Enter the number whose bits need to be swapped:-\n");
+    scanf("%u", &x);
+    printf("%u ", swapBits(x));
+    return 0;
+}
+
+/* Time Complexity:- O(1)
+   Space Complexity:- O(1)
+*/
```

Comment:
Add sample I/O.

Reply:
Yes, I have done it. Now, you can review my code again.

Reply:
No. You have not resolved it yet